Gradle build and clean tests

subhash-arabhi opened the pull request:
1. Added tests to see how Java workspace compile and clean works on Gradle projects
2. Reorganised the test folder into different suites 
3. Added few utility functions for testing

Achal1607 reviewed (changes requested):
nitpick: There are some linting issues, can you please format the code. Also, after doing this change, can you please squash the commits.

subhash-arabhi commented:
> nitpick: There are some linting issues, can you please format the code. Also, after doing this change, can you please squash the commits.

Fixed linting issues and squashed main commits

Achal1607 commented:
Tests are failing, can you please delete `package-lock.json` and  run `npm i` again.
